txrelayer/txrelayer.go

```diff
@@ -86,6 +87,14 @@ func (t *TxRelayerImpl) Call(from ethgo.Address, to ethgo.Address, input []byte)
 func (t *TxRelayerImpl) SendTransaction(txn *ethgo.Transaction, key ethgo.Key) (*ethgo.Receipt, error) {
 	txnHash, err := t.sendTransactionLocked(txn, key)
 	if err != nil {
+		if txn.Type != ethgo.TransactionLegacy && errors.Is(err, types.ErrTxTypeNotSupported) {
```

Comment:
This part is a fallback from if blockchain does not accept dynamic transactions and we check if error is `ErrTxTypeNotSupported`. How do we know if some rootchain is going to return us this error if it does not support EIP-1559?

Reply:
We cannot be sure, but this was the best guess. Geth returns this error for instance, although it is wrapped (https://github.com/ethereum/go-ethereum/blob/master/core/txpool/validation.go#L63-L65). Bor returns this exact error message, since it is a fork of Geth (https://github.com/maticnetwork/bor/blob/develop/core/txpool/txpool.go#L754-L756).

So I'd say we have two options here:
- if any error is returned, try with `LegacyTx` fallback (though not sure is it ok to do it),
- if error message contains `ErrTxTypeNotSupported` string, fallback to `LegacyTx`

What do you think?

Reply:
Leave it as it is and add another task into the backlog to check this.